src/Sylius/Component/Core/Model/OrderItem.php

```diff
@@ -57,4 +57,45 @@ public function equals(BaseOrderItemInterface $item)
             && $item->getVariant() === $this->variant
         );
     }
+
+    /**
+     * Returns sum of neutral and non neutral tax adjustments on order item and total tax of units.
+     *
+     * {@inheritdoc}
+     */
+    public function getTaxTotal()
+    {
+        $taxTotal = 0;
+
+        foreach ($this->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT) as $taxAdjustment) {
+            $taxTotal += $taxAdjustment->getAmount();
+        }
+
+        foreach ($this->units as $unit) {
+            $taxTotal += $unit->getTaxTotal();
+        }
+
+        return $taxTotal;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDiscountedUnitPrice()
+    {
+        $orderItemDiscounts = 0;
+        foreach ($this->units as $unit) {
+            $orderItemDiscounts += $unit->getAdjustmentsTotal(AdjustmentInterface::ORDER_ITEM_PROMOTION_ADJUSTMENT);
```

Comment:
Surely there is a bug, as ``ORDER_ITEM_PROMOTION``s are distributed on units, we should not take whole this discount to count ``discountedUnitPrice``... However, there is a problem, as in specific cases promotions distributed on units can be not equal. We'll rethink how to solve this issue ;)